doc/source/serve/http-guide.md

```diff
@@ -95,88 +117,148 @@ def f():
+
+```bash
+ray start --head
+python main.py
+```
+
+(serve-http-adapters)=
+
+## HTTP Adapters
+
+HTTP adapters are functions that convert raw HTTP requests to basic Python types that you know and recognize.
+
+For example, here is an adapter that extracts the JSON content from a request:
 
 ```python
-import ray
+async def json_resolver(request: starlette.requests.Request):
+    return await request.json()
+```
 
-from fastapi import FastAPI
-from ray import serve
+The input arguments to an HTTP adapter should be type-annotated. At a minimum, the adapter should accept a `starlette.requests.Request` type (https://www.starlette.io/requests/#request),
+but it can also accept any type that's recognized by [FastAPI's dependency injection framework](https://fastapi.tiangolo.com/tutorial/dependencies/).
 
-app = FastAPI()
-ray.init(address="auto", namespace="summarizer")
-serve.start(detached=True)
+Here is an HTTP adapter that accepts two HTTP query parameters:
 
-@app.get("/")
-def f():
-    return "Hello from the root!"
+```python
+def parse_query_args(field_a: int, field_b: str):
+    return YourDataClass(field_a, field_b)
+```
 
-@serve.deployment(route_prefix="/api1")
-@serve.ingress(app)
-class FastAPIWrapper1:
-    @app.get("/subpath")
-    def method(self):
-        return "Hello 1!"
+You can specify different type signatures to facilitate the extraction of HTTP fields, including
+- [query parameters](https://fastapi.tiangolo.com/tutorial/query-params/),
+- [body parameters](https://fastapi.tiangolo.com/tutorial/body/),
+and 
+- [many other data types](https://fastapi.tiangolo.com/tutorial/extra-data-types/).
 
-@serve.deployment(route_prefix="/api2")
-@serve.ingress(app)
-class FastAPIWrapper2:
-    @app.get("/subpath")
-    def method(self):
-        return "Hello 2!"
+For more details, you can take a look at the [FastAPI documentation](https://fastapi.tiangolo.com/).
+
+You can use adapters in different scenarios within Serve, which we will go over one by one:
+
+- Ray AIR `Predictor`
+- Serve Deployment Graph `DAGDriver`
+- Embedded in Bring Your Own `FastAPI` Application
+
+### Ray AIR `Predictor`
+
+Ray Serve provides a suite of adapters to convert HTTP requests to ML inputs like `numpy` arrays.
+You can use them together with the [Ray AI Runtime (AIR) model wrapper](air-serving-guide) feature
+to one-click deploy pre-trained models.
+
+As an example, we provide a simple adapter for an *n*-dimensional array.
 
-FastAPIWrapper1.deploy()
-FastAPIWrapper2.deploy()
+When using [model wrappers](air-serving-guide), you can specify your HTTP adapter via the `http_adapter` field:
+
+```python
+from ray import serve
+from ray.serve.http_adapters import json_to_ndarray
+from ray.serve import PredictorDeployment
+
+serve.run(PredictorDeployment.options(name="my_model").bind(
+    my_ray_air_predictor,
+    my_ray_air_checkpoint,
+    http_adapter=json_to_ndarray
+))
```

Comment:
Do you want to mention what `my_air_predictor`, `my_ray_air_checkpoint` are as arguments to the .`bind()` call?